test/e2e/storage/testsuites/provisioning.go

```diff
@@ -225,6 +225,26 @@ func (p *provisioningTestSuite) defineTests(driver TestDriver, pattern testpatte
 		}
 		l.testCase.TestDynamicProvisioning()
 	})
+	ginkgo.It("should provision storage with pvc data source [Feature:VolumePVCDataSource]", func() {
+		if !dInfo.Capabilities[CapPVCDataSource] {
+			framework.Skipf("Driver %q does not support cloning - skipping", dInfo.Name)
+		}
+
+		init()
+		defer cleanup()
+
+		dc := l.config.Framework.DynamicClient
+		dataSource, dataSourceCleanup := preparePVCDataSourceForProvisioning(framework.NodeSelection{Name: l.config.ClientNodeName}, l.cs, dc, l.pvc, l.sc)
+		defer dataSourceCleanup()
+
+		l.pvc.Spec.DataSource = dataSource
```

Comment:
Have you tried running this test manually against hostpath driver?  I'm not sure if reusing `l.pvc` works here because we already used it to provision the source volume.

Reply:
I've confirmed that works, we're just basically adjusting the spec for the pvc we want to create by appending the DataSource in to it.  That being said, the one issue is that the e2e tests still seem to be using an old version of the csi provisioner, as such the e2e test fails on this for an invalid DS type.

Reply:
The e2e test is still deploying v1.1.0 of the csi-external-provsioner, so this test fails due to the obsolete DataSource type check.

Reply:
You're right, this isn't quite right; working on a fix now.

Reply:
I'm still not sure this works. `l.pvc` is used to create the source pvc.  We need a new PVC for the target clone.

Reply:
I'll clean it up, it's poorly written as is even if it did work.

Reply:
@msau42 I added a new source PVC to the init, instrumented some thing and ran though it with the new images to confirm it's working as expected.